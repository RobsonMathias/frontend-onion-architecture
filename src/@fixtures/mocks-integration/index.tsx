export * from './store.mock';
export * from './adapters.mock';
export * from './../mocks/http.mock';
export * from './../mocks/router.mock';
