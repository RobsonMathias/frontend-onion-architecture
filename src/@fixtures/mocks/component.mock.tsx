import React from 'react';

export function Mock() {
  return <div />;
}

export const mockComponent = (componentsName: string[] = []) => {
  let composed: { [key: string]: React.ReactNode } = {};
  componentsName.forEach((name: string) => {
    composed[name] = Mock;
  });
  return composed;
};
