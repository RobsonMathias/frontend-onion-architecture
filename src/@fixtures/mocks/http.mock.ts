export const mockHttp = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  interceptors: {
    request: {
      use: jest.fn(),
      eject: jest.fn()
    },
    response: {
      use: jest.fn()
    }
  }
} as any;
