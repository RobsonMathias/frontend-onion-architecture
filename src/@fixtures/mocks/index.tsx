export * from './store.mock';
export * from './router.mock';
export * from './component.mock';
export * from './wrappers';
export * from './http.mock';
export * from './services.mock';
export * from './navigation.mock';
export * from './storage.mock';
export * from './use-auth.mock';
