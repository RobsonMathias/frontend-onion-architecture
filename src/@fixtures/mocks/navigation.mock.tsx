import React from 'react';

const setData = jest.fn();
const mockUseUI = jest.fn(() => ({ setData }));

const query = { get: jest.fn(), set: jest.fn() };
const set = jest.fn();
const setAll = jest.fn();
const mockUseQuery = jest.fn(() => ({ query, setAll, set }));

const replace = jest.fn();
const mockUseURL = jest.fn(() => ({ replace }));

const mockNavBar = React.forwardRef((props, ref) => <div {...props} />);

export const mockNavigation = {
  NavBar: mockNavBar,
  useQuery: mockUseQuery,
  query,
  setAll,
  set,
  useUI: mockUseUI,
  setData,
  useURL: mockUseURL,
  replace
};
