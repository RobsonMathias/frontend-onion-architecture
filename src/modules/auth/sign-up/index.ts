export * from './sign-up.container';
