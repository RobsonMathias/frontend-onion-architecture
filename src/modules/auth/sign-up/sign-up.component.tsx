import React, { useEffect } from 'react';

import { SignUpPayload } from '@types';
import { Box, Button, TextField } from '@mui/material';
import { FieldErrors, useForm } from 'react-hook-form';

type SignUpComponentProps = {
  onSubmit: (payload: SignUpPayload) => void;
  loading?: boolean;
};

export function SignUpComponent(props: SignUpComponentProps) {
  const { onSubmit, loading } = props;
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors, isSubmitted }
  } = useForm({ mode: 'all' });

  useEffect(() => {
    return () => {
      reset();
    };
  }, []);

  function hasError(error?: FieldErrors): string {
    const { message } = error || { message: '' };
    return isSubmitted ? message : '';
  }

  function submit(data: SignUpPayload) {
    onSubmit(data);
  }

  return (
    <form onSubmit={handleSubmit(submit)}>
      <Box mb={2}>
        <TextField
          variant="standard"
          fullWidth={true}
          type="email"
          inputProps={{ 'data-testid': 'email' }}
          label="E-mail"
          error={!!hasError(errors.email)}
          helperText={hasError(errors.email)}
          placeholder="email@exemplo.com"
          {...register('email', {
            required: 'E-mail é obrigatório.'
          })}
        />
      </Box>
      <Box mb={2}>
        <TextField
          variant="standard"
          fullWidth={true}
          type="text"
          inputProps={{ 'data-testid': 'name' }}
          label="Nome"
          placeholder="Digite seu nome"
          error={!!hasError(errors.name)}
          helperText={hasError(errors.name)}
          {...register('name', {
            required: 'Nome é obrigatório.'
          })}
        />
      </Box>
      <Box mb={2}>
        <TextField
          type="password"
          variant="standard"
          inputProps={{ 'data-testid': 'password' }}
          fullWidth={true}
          autoComplete="off"
          label="Senha"
          placeholder="Digite sua senha"
          error={!!hasError(errors.password)}
          helperText={hasError(errors.password)}
          {...register('password', {
            minLength: {
              value: 9,
              message: 'Senha precisa ter no mínimo 9 caracteres.'
            },
            required: 'Senha é obrigatória.'
          })}
        />
      </Box>
      <Button
        disabled={loading}
        data-testid="submit"
        type="submit"
        variant="contained"
        color="secondary"
      >
        Cadastrar
      </Button>
    </form>
  );
}
