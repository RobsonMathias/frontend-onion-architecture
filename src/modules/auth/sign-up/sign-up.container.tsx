import React from 'react';

import { SignUpComponent } from './sign-up.component';
import { useSignUp } from '../hooks';
import { STORE_STATUS_LOADING } from 'constants/store';
import { SignUpPayload } from '@types';

export function SignUpContainer() {
  const { signUp, user } = useSignUp();

  function onSubmit(payload: SignUpPayload) {
    signUp(payload);
  }

  return <SignUpComponent onSubmit={onSubmit} loading={user.status === STORE_STATUS_LOADING} />;
}
