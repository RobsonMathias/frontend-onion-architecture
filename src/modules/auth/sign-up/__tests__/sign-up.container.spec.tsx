/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { mockComponent } from '@fixtures/mocks';
import { render } from '@testing-library/react';
import React from 'react';
import {
  mockAuthStore,
  STORE_AUTH_FETCH_USER_MOCK_FINISHED as mockAuthUserStore
} from '@fixtures/stores/auth';

jest.mock('../../hooks', () => ({
  useSignUp: jest.fn(() => ({ ...mockAuthStore.auth, ...mockAuthUserStore }))
}));
jest.mock(__dirname + '/../sign-up.component', () => mockComponent(['SignUpComponent']));

import { SignUpContainer } from '../sign-up.container';

describe('SignUp', () => {
  it('should render successfully', () => {
    const { container } = render(<SignUpContainer />);
    expect(container).toMatchSnapshot();
  });
});
