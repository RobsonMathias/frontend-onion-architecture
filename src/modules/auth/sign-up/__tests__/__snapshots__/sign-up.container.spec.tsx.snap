// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render successfully 1`] = `
<div>
  <div />
</div>
`;
